crates/volta-core/src/toolchain/mod.rs

```diff
@@ -52,37 +52,38 @@ impl Toolchain {
                 file: path.to_owned(),
             })?;
 
-        let platform = serial::Platform::from_json(src)?.into_platform()?;
+        let platform = serial::Platform::from_json(src)?
+            .into_platform()?
+            .map(Rc::new);
         if platform.is_some() {
             debug!("Found default configuration at '{}'", path.display());
         }
         Ok(Toolchain { platform })
     }
 
-    pub fn platform_ref(&self) -> Option<&PlatformSpec> {
-        self.platform.as_ref()
+    pub fn platform(&self) -> Option<Rc<PlatformSpec>> {
+        self.platform.clone()
     }
 
     /// Set the active Node version in the default platform file.
-    pub fn set_active_node(&mut self, node_version: &NodeVersion) -> Fallible<()> {
+    pub fn set_active_node(&mut self, node_version: &Version) -> Fallible<()> {
         let mut dirty = false;
 
-        if let Some(ref mut platform) = self.platform {
-            if platform.node_runtime != node_version.runtime {
-                platform.node_runtime = node_version.runtime.clone();
-                dirty = true;
-            }
-
-            if platform.npm.as_ref() != Some(&node_version.npm) {
-                platform.npm = Some(node_version.npm.clone());
+        if let Some(platform) = &self.platform {
+            if platform.node.version != *node_version {
```

Comment:
As a point of curiosity: any reason to prefer `platform.node.version != *node_version` over `&platform.node.version != node_version`?

Reply:
None really, other than that conceptually it makes more sense in my head to compare the values (by dereferencing the borrow) than the references. I know that both result in the same comparison, so if the borrow approach is more idiomatic Rust then I'm happy to swap.

Reply:
I don't think there's any particular preference one way or the other – I personally tend to reach for borrow over deref, but I think it's totally fine either way! Was just curious what *your* feeling was, mostly!